themes/default/content/blog/pulumi-release-notes-m64/index.md

```diff
@@ -0,0 +1,234 @@
+
+[Learn more about AWS auto-naming in this GitHub issue](https://github.com/pulumi/pulumi-aws-native/issues/156)
+
+### New resources in the AWS Native provider
+
+We shipped new versions of the AWS Native provider (0.3.0 through 0.4.0) that added [38 new resources](https://github.com/pulumi/pulumi-aws-native/compare/v0.2.0...v0.4.0#diff-1ac835cc58d7899e9299c7570151c7b0d7732c78f1bd53fe25fd4189b72e168e) giving you access to resources like `AWS::EKS::Cluster`, `AWS::Lightsail::Database`, and several new resources in the `AWS::Redshift` namespace.
+
+### New resources in the Azure Native provider
+
+We shipped new versions of the Azure Native provider (1.43.0 through 1.45.0) that collectively added [4 new resources](https://github.com/pulumi/pulumi-azure-native/blob/master/CHANGELOG.md#1450-2021-11-05) giving you access to resources like Cognitive Services.
+
+## Pulumi CLI and core technologies
+
+In this milestone, we shipped Pulumi versions [3.16.1](https://github.com/pulumi/pulumi/releases/tag/v3.16.1) through [3.17.0](https://github.com/pulumi/pulumi/releases/tag/v3.17.0). The full list of changes in each version is available in the [changelog](https://github.com/pulumi/pulumi/blob/master/CHANGELOG.md); read on to learn about some of the biggest changes.
+
+### Functions now support outputs
+
+When creating complex infrastructure, it's common to pass information about one resource to a function provided by another resource. Previously, doing this in Pulumi required you to use `apply` to get the information from the first resource and pass it into the function. Now, we've simplified the process so you can reference outputs directly from the function—no `apply` needed. To take advantage of these changes, make sure you've updated to the latest versions of Pulumi and the providers you're using.
+
+[Learn more in the "Functions now accept outputs" blog post]({{<relref "/blog/functions-accept-outputs">}})
+
+### `--json` flag for `pulumi {update|destroy|preview|refresh}`
+
+When automating or troubleshooting Pulumi operations, it can be helpful to see all of the details emitted by the Pulumi engine rather than the standard progress bar. Now, you can! Just add `--json` to your `pulumi {update|destroy|preview|refresh}` command.
+
+[Learn more about the new `--json` flag in this GitHub issue](https://github.com/pulumi/pulumi/issues/2390)
+
+### Python 3.6 support will be deprecated soon
+
+Python 3.6 will be end-of-lifed ("EOLed") on [December 23, 2021](https://www.python.org/downloads/#:~:text=2016-12-23-,2021-12-23,-PEP%20494). Around that time, Pulumi will drop support for Python 3.6. If you're using Python 3.6, consider upgrading to a newer version of Python. The Pulumi CLI will display a message when it runs a Pulumi program using Python 3.6.
```

Comment:
I wish we had a "standard EOL" policy we could link to here. And I would love if that EOL policy was not "right around the time the tech itself is EOL," but instead gives a quarter to two of grace period. Many enterprises have a tough time migrating and, unless we're saving ourselves a significant cost, not clear extra pain is worthwhile.

Reply:
I've passed this feedback onto the team - @emiliza is working on an updated policy we can post for each language.